pkg/asset/installconfig/azure/client.go

```diff
@@ -152,3 +156,31 @@ func (c *Client) getProvidersClient(ctx context.Context) (azres.ProvidersClient,
 	client.Authorizer = c.ssn.Authorizer
 	return client, nil
 }
+
+// ValidateResourceSkuAvailability checks if the diskType exists for the given region.
+func (c *Client) ValidateResourceSkuAvailability(ctx context.Context, region string, diskType string) (*azsku.ResourceSku, error) {
+	client := azsku.NewResourceSkusClient(c.ssn.Credentials.SubscriptionID)
+	client.Authorizer = c.ssn.Authorizer
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+
+	resourceSkus, err := client.List(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for !resourceSkus.NotDone() {
+		for _, page := range resourceSkus.Values() {
+			if to.String(page.Name) == diskType {
+				for _, diskRegion := range to.StringSlice(page.Locations) {
+					if diskRegion == region {
+						return &page, nil
+					}
+				}
+			}
+		}
+		resourceSkus.Next()
+	}
```

Comment:
i would like to keep the code for paging azure apis consistent, like
https://github.com/openshift/installer/blob/2672f9a16e260c1134476c83a384e0d030185f26/pkg/asset/machines/azure/zones.go#L44
or
https://github.com/openshift/installer/blob/2672f9a16e260c1134476c83a384e0d030185f26/pkg/asset/installconfig/azure/dns.go#L148
```suggestion
	for skuPage, err := client.List(ctx); skuPage.NotDone(); err = zonesPage.NextWithContext(ctx) {
		for _, page := range skuPage.Values() {
			if to.String(page.Name) == diskType {
				for _, diskRegion := range to.StringSlice(page.Locations) {
					if strings.EqualFold(diskRegion, region) {
						return &page, nil
					}
				}
			}
		}
	}
```